lisp/pdf-view.el

```diff
@@ -1219,6 +1220,45 @@ The colors are determined by the variable
   (pdf-cache-clear-images)
   (pdf-view-redisplay t))
 
+(defun pdf-view-refresh-all-themed-buffers (&optional args)
+  "Ensure all PDFView buffers use the new theme for their
+background and background colors."
+  (mapc (lambda (buffer)
+          (with-current-buffer buffer
+            (when pdf-view-themed-minor-mode
+              (pdf-cache-clear-images)
+              (pdf-view-set-theme-background)
+              (pdf-view-redisplay t))))
+        (buffer-list)) nil)
+
+(defun pdf-view-set-theme-background ()
+  (pdf-tools-assert-pdf-buffer)
+  (pdf-info-setoptions
+   :render/foreground (face-foreground 'default nil)
+   :render/background (face-background 'default nil)
+   :render/usecolors t))
+
+(define-minor-mode pdf-view-themed-minor-mode
+  "Apply a color-filter to all pdf-view buffers that keeps synchronized with the Emacs theme.
+
+The colors are determined by the `face-foreground' and
+`face-background' of the currently active theme."
+
+  nil " Thm" nil
+  ;; FIXME: Maybe these options should be passed stateless to pdf-info-renderpage ?
+    (cond
+     (pdf-view-themed-minor-mode
+      (pdf-view-set-theme-background)
+      (add-hook 'after-save-hook #'pdf-view-set-theme-background nil t)
+      (add-hook 'after-revert-hook #'pdf-view-set-theme-background nil t)
+      (advice-add 'enable-theme :after #'pdf-view-refresh-all-themed-buffers))
+     (t
+      (remove-hook 'after-save-hook #'pdf-view-set-theme-background t)
+      (remove-hook 'after-revert-hook #'pdf-view-set-theme-background t)
+      (advice-remove 'enable-theme #'pdf-view-refresh-all-themed-buffers)
+      (pdf-info-setoptions :render/usecolors nil)))
+    (pdf-view-refresh-all-buffers))
```

Comment:
This function does not exist.

Please run `make check` before committing and pay attention to the compiler warnings.